src/schema.cpp

```diff
@@ -28,29 +28,30 @@
 using namespace realm;
 
 namespace realm {
-bool operator==(Schema const& a, Schema const& b)
+bool operator==(Schema const& a, Schema const& b) noexcept
 {
     return static_cast<Schema::base const&>(a) == static_cast<Schema::base const&>(b);
 }
 }
 
-Schema::Schema() = default;
+Schema::Schema() noexcept = default;
 Schema::~Schema() = default;
 Schema::Schema(Schema const&) = default;
-Schema::Schema(Schema &&) = default;
+Schema::Schema(Schema &&) noexcept = default;
 Schema& Schema::operator=(Schema const&) = default;
-Schema& Schema::operator=(Schema&&) = default;
+Schema& Schema::operator=(Schema&&) noexcept = default;
 
 Schema::Schema(std::initializer_list<ObjectSchema> types) : Schema(base(types)) { }
 
-Schema::Schema(base types) : base(std::move(types))
+Schema::Schema(base types) noexcept
+: base(std::move(types))
 {
     std::sort(begin(), end(), [](ObjectSchema const& lft, ObjectSchema const& rgt) {
```

Comment:
I'm not convinced about the casual use of `noexcept` everywhere. What if `std::sort` runs out of memory and throws a `bad_alloc`? Do we want to change the behaviour in this scenario from propagating the exception up to the SDKs, to an abort? I doubt the SDK's have any sort of fancy error recovery in such a scenario, but I do want to make sure this change in behaviour is intentional.

Reply:
std::sort is not allowed to throw any exceptions when given valid input and a comparison function which does not throw, which is normally achieved by using a sort algorithm that does not need to allocate any non-stack scratch space.

Reply:
I couldn't find anything about that in the C++11 standard, doesn't that mean it is implementation dependant? I'm just curious where I can find the information, I don't mean to belabour it too much (not-blocking).

Reply:
The standard explicitly lists all exceptions that an algorithm can throw. Since none are listed for std::sort(), that means it's not allowed to throw any itself (but it's not noexcept because the predicate might throw).